### PR TITLE
Ensure that a broken OCI spec does not break inspect

### PR DESCRIPTION
**Pull request by mheon**
The process of saving the OCI spec is not particularly reboot-safe. Normally, this doesn't matter, because we recreate the spec every time a container starts, but if one was to reboot (or SIGKILL, or otherwise fatally interrupt) Podman in the middle of writing the spec to disk, we can end up with a malformed spec that sticks around until the container is next started. Some Podman commands want to read the latest version of the spec off disk (to get information only populated after a container is started), and will break in the case that a partially populated spec is present. Swap to just ignoring these errors (with a logged warning, to let folks know something went wrong) so we don't break important commands like `podman inspect` in these cases.

[NO NEW TESTS NEEDED] Provided reproducer involves repeatedly rebooting the system

```release-note
NONE
```

### CODE REVIEW

**Comment by mheon** [comment]
Should go green now, @containers/podman-maintainers PTAL

**Comment by Luap99** [comment]
Will this cause panics when the spec is nil?

**Comment by mheon** [comment]
We're returning `c.config.Spec` which is still a valid spec - just not the invalid one from disk.

**Review by Luap99 — COMMENTED** [review]
LGTM, but I wonder if we should remove the spec from disc if it is invalid to avoid spamming this message?

**Comment by mheon** [comment]
Hm. We'd have to check if the container is running first, I imagine? Removing an invalid OCI spec while the container does not exist seems perfectly valid; removing an invalid OCI spec that has somehow been used to create a valid, running container seems like a bad idea.

**Review by giuseppe — APPROVED** [review]
/lgtm

**Comment by mheon** [comment]
Bot is broklen, doing it manually

**Pull request merged**
